app/frontend/src/authConfig.ts

```diff
@@ -156,3 +157,31 @@ export const getToken = (client: IPublicClientApplication): Promise<string | und
             return undefined;
         });
 };
+
+export const getUsername = async (client: IPublicClientApplication): Promise<string | null> => {
```

Comment:
I wonder if itd be helpful to add docstrings like "Gets the username from either MSAL.JS-based authentication or App Service built-in auth" to help developers understand why we have all these functions. That could also be a top-level comment (with links to relevant doc overviews), but for intellisense, it can be nice to have function docstrings too.